packages/queued-request-controller/tests/QueuedRequestController.test.ts

```diff
@@ -0,0 +1,224 @@
+import type { QueuedRequestControllerOptions } from '../src';
+import { QueuedRequestController } from '../src';
+import { buildQueuedRequestControllerMessenger } from './utils';
+
+describe('QueuedRequestController', () => {
+  it('can be instantiated with default values', () => {
+    const options: QueuedRequestControllerOptions = {
+      messenger: buildQueuedRequestControllerMessenger(),
+    };
+
+    const controller = new QueuedRequestController(options);
+    expect(controller.state).toStrictEqual({});
+  });
+
+  describe('enqueueRequest', () => {
+    it('runs the next request immediately when the queue is empty', async () => {
+      const options: QueuedRequestControllerOptions = {
+        messenger: buildQueuedRequestControllerMessenger(),
+      };
+
+      const controller = new QueuedRequestController(options);
+
+      // Mock requestNext function
+      const requestNext = jest.fn(() => Promise.resolve());
+
+      await controller.enqueueRequest(requestNext);
+
+      // Expect that the request was called
+      expect(requestNext).toHaveBeenCalledTimes(1);
+    });
+
+    it('runs each request sequentially in the correct order', async () => {
+      const options: QueuedRequestControllerOptions = {
+        messenger: buildQueuedRequestControllerMessenger(),
+      };
+
+      const controller = new QueuedRequestController(options);
+
+      // Mock requestNext function with resolved promises
+      // Use an array to track the order of execution
+      const executionOrder: string[] = [];
+
+      // Enqueue requests
+      controller.enqueueRequest(async () => {
+        executionOrder.push('Request 1 Start');
+        await new Promise((resolve) => setTimeout(resolve, 10));
+        executionOrder.push('Request 1 End');
+      });
+
+      await controller.enqueueRequest(async () => {
+        executionOrder.push('Request 2 Start');
+        await new Promise((resolve) => setTimeout(resolve, 10));
+        executionOrder.push('Request 2 End');
+      });
+
+      // Assert that the execution order is correct
+      expect(executionOrder).toStrictEqual([
+        'Request 1 Start',
+        'Request 1 End',
+        'Request 2 Start',
+        'Request 2 End',
+      ]);
+    });
+
+    describe('error handling', () => {
+      it('handles errors when a request fails', async () => {
+        const options: QueuedRequestControllerOptions = {
+          messenger: buildQueuedRequestControllerMessenger(),
+        };
+
+        const controller = new QueuedRequestController(options);
+
+        // Mock a request that throws an error
+        const requestWithError = jest.fn(() =>
+          Promise.reject(new Error('Request failed')),
+        );
+
+        // Enqueue the request
+        try {
+          await controller.enqueueRequest(requestWithError);
```

Comment:
this won't fail if enqueueRequest doesn't throw. Should explicitly expect an error

```
await expect(controller.enqueueRequest(requestWithError).rejects.toThrowError('Request failed')
expect(controller.length()).toBe(0)
```

Reply:
same issue for the other try catches that have an expect in the catch